Add release notes for Trino Gateway 10 (#411)
trinodb/trino-gateway: docs/release-notes.md

# Release notes

> ## Trino Gateway 10 (?? July 2024)

Reviewer: ```suggestion
## Trino Gateway 10 (23 July 2024)
```

[JAR file gateway-ha-10-jar-with-dependencies.jar](https://repo1.maven.org/maven2/io/trino/gateway/gateway-ha/10/gateway-ha-10-jar-with-dependencies.jar),
Docker container `trinodb/trino-gateway:10`

> * [:warning: Breaking change:](#breaking) Remove support for any Dropwizard and

Reviewer: ```suggestion
* [:warning: Breaking change:](#breaking) Remove support for Dropwizard and
```

  Jetty Proxy integration and usage. Add

>   [Airlift](https://github.com/airlift/airlift) as base application framework as

Reviewer: ```suggestion
  [Airlift](https://github.com/airlift/airlift) as the base application framework
```

  used in Trino. This changes the supported Trino Gateway startup, configuration

>   files, and relevant properties. Find details in the documentation, and

Reviewer: Link to documentation on "Find details in the documentation?"

  specifically refer to the [upgrade guide](migration-to-airlift.md) when
  migrating from older releases.
  ([#41](https://github.com/trinodb/trino-gateway/issues/41))

> * [:warning: Breaking change:](#breaking) Improve Helm chart reliability and

Reviewer: "adjust to new Airlift base framework" doesn't feel like it means much from a user-facing perspective, why is this a breaking change?

Author: because any custom modules are breaking and any existing config that relies on dropwizard (and there is a whole bunch) also breaks.

  adjust to new Airlift base framework.
  ([#401](https://github.com/trinodb/trino-gateway/pull/401))
* Enable routing rules to use values extracted from the HTTP request.
  ([#325](https://github.com/trinodb/trino-gateway/pull/325))

> * Add support for using OIDC claim for authorization.

Reviewer: ```suggestion
* Add support for using an OIDC claim for authorization.
```

# Release notes

## Trino Gateway 10 (?? July 2024)

[JAR file gateway-ha-10-jar-with-dependencies.jar](https://repo1.maven.org/maven2/io/trino/gateway/gateway-ha/10/gateway-ha-10-jar-with-dependencies.jar),
Docker container `trinodb/trino-gateway:10`

* [⚠](#breaking) Remove support for any Dropwizard and Jetty Proxy integration 
  and usage. Add [Airlift](https://github.com/airlift/airlift) as base 
  application framework as used in Trino. This changes the supported Trino 
  Gateway configuration and relevant properties. Find details in the
  documentation, and specifically refer to the
  [upgrade guide](migration-to-airlift.md) when migrating from older releases. 

>   ([#41](https://github.com/trinodb/trino-gateway/issue/41)),

Reviewer 2: Broken links. `issue` -> `issues` for all the links to issue.

  ([#317](https://github.com/trinodb/trino-gateway/pull/317)),

>   ([#382](https://github.com/trinodb/trino-gateway/pull/382)),

Reviewer 2: I think simply linking to the issue is sufficient.

Author: I removed the two PR links and added a comment in the issue instead.

* Support reading logging configuration from properties file. 

>   ([#340](https://github.com/trinodb/trino-gateway/issue/340))

Reviewer 2: This feature being overwritten by the Airlift changes.

Author: So we drop it? Its still possible now right .. ?

Author: I will drop from release notes.

# Release notes

## Trino Gateway 10 (24 July 2024)

[JAR file gateway-ha-10-jar-with-dependencies.jar](https://repo1.maven.org/maven2/io/trino/gateway/gateway-ha/10/gateway-ha-10-jar-with-dependencies.jar),

> Docker container `trinodb/trino-gateway:10`

Reviewer 3: I know that this is outside of the scope for this PR but I think that we should rename docker image to just `trinodb/gateway`

Author: no .. we agreed in the past to use Trino Gateway as project and product name consistently. This is also in line with naming things like the Trino CLI, the Trino Go Client and so on..

* [:warning: Breaking change:](#breaking) Remove support for Dropwizard and
  Jetty Proxy integration and usage. Add
  [Airlift](https://github.com/airlift/airlift) as the base application

>   framework as

Reviewer 4: NIT: the line break, not sure how it would look?

Reviewer 3: This is an implementation detail so I'm not sure whatever we should call it out. Configuration changes obviously but not framework itself

Author: I think it is important since it affects custom module writing and other aspects quite a lot as well and our audience is currently not just administrators running Trino Gateway but also heavily developers modifying it.

Author: Rendering works fine.

  used in Trino. This changes the supported Trino Gateway startup, configuration
  files, and relevant properties. Find details in the
  [documentation](quickstart.md), and specifically refer to the
  [upgrade guide](migration-to-airlift.md) when migrating from older releases.
  ([#41](https://github.com/trinodb/trino-gateway/issues/41))
* [:warning: Breaking change:](#breaking) Improve Helm chart reliability and
  adjust to new Airlift base framework.
  ([#401](https://github.com/trinodb/trino-gateway/pull/401))

> * Enable routing rules to use values extracted from the HTTP request.

Reviewer 5: ```suggestion
* Enable routing rules to use query and user details extracted from the HTTP request.
```

* Remove concurrency issue from repeated rules file loading. ([#9](https://github.com/trinodb/trino-gateway/pull/9))

[Details about all merged pull requests](https://github.com/trinodb/trino-gateway/pull/52)

## Breaking changes <a name="breaking">

Starting with Trino Gateway 10, release notes include a ⚠️ symbol to 
highlight any changes as potentially breaking changes. The following 
changes are considered and may require adjustments:

* Removal or renaming of configuration properties that may prevent startup or 
require configuration changes.
* Changes to default values for configuration properties that may 
  significantly change the behavior of a system.
* Updates to the requirements for external systems or software used with 

>   Trino Gateway, such as removal of support for an old version of a database 

Reviewer 6: removal of support for an old version of a database for the database sounds redundant... maybe just removal of support for an old version of the database?

Author: Removed.
